src/raw_device.rs

```diff
@@ -101,18 +110,17 @@ impl<SpiBus: Bus> RawDevice<SpiBus> {
             frame
         };
 
-        let received_frame_size = self.read_bytes(read_buffer, 2)?;
+        let received_frame_size = self.read_bytes(
+            read_buffer,
+            2,
+            expected_frame_size.saturating_sub(read_buffer.len()) as u16,
```

Comment:
I think the right approach would be to actually add a new `self.skip_bytes()` function that intentionally fast-forwards the read pointer to the end of the frame if the `read_buffer` is too long. As written, the logic for fast forwarding is hard for me to follow.